src/lib/server/db/bans.channels.users.sql.ts

```diff
@@ -0,0 +1,28 @@
+import { boolean, pgTable, timestamp, unique, uuid } from 'drizzle-orm/pg-core';
+import { channel } from './channels.sql';
+import { user } from './users.sql';
+
+export const channelBannedUser = pgTable(
+    'channel_user_ban',
+    {
+        id: uuid('id').primaryKey().defaultRandom(),
+        channelId: uuid('channel_id')
+            .notNull()
+            .references(() => channel.id),
+        userId: uuid('user_id')
+            .notNull()
+            .references(() => user.id),
+        createdOn: timestamp('created_on').notNull().defaultNow(),
+        pardoned: boolean('pardoned').notNull().default(false),
+    },
+    (table) => ({ unq: unique().on(table.channelId, table.userId) })
```

Comment:
I see a problem with this uniqueness constraint when a user is banned, then pardoned, then banned again.

Reply:
Totally right. I'll take another look at this